Feature/test empty db

delcroip opened the pull request:
- Mostly flake8
- Model.filter_validity
- ORM based validations

delcroip committed:
turn validaiton to ORM

delcroip committed:
chore: less db calls

delcroip committed:
chore:37 sec

delcroip committed:
chore: less db calls

delcroip committed:
feat: activate max

delcroip committed:
chore: removing more db calls

delcroip committed:
chore: missing import

delcroip committed:
chore: flake8

delcroip committed:
chore: flake ignore reports

delcroip committed:
chore: add test ynote case

delcroip committed:
chore: use insert_role_right_for_system

delcroip committed:
Merge remote-tracking branch 'origin/feature/validation_orm' into feature/test-empty-db

delcroip committed:
refactor(test): improve claim validation tests with helpers

Replace direct database queries with test helper functions (e.g., create_test_health_facility, create_test_insuree) to enhance test isolation and reliability. Rename test method from 'test_validate_polivx' to 'test_validate_policy' for clarity. Add 'is_process' parameter to validate_claim call in claim processing. Minor comment formatting fixes in attachment creation.

delcroip committed:
feat(claim): add processing timestamps for claims

Add process_stamp and date_processed fields to claim objects when status is set to PROCESSED, recording the exact datetime and date of processing for better tracking and auditing. Includes necessary import for datetime module.

delcroip committed:
refactor: standardize filter_validity calls to model class methods

Replace standalone filter_validity() function calls with Model.filter_validity()
across various GraphQL mutations in claim/gql_mutations.py to improve encapsulation
and consistency in querying valid records for Claim, ClaimAttachment, ClaimItem,
ClaimService, Service, and Item models.

delcroip committed:
feat: remove unused filter_validity import from claim module

Remove unused `filter_validity` import from multiple files in the claim module to clean up the codebase. This change improves code maintainability by removing unnecessary dependencies.

sunilparajuli commented:
@delcroip this is required , i still cannot test in postgres , throws error for test db. I can share full error log if required but , how about testing with test json, test objects, etc. similar to simulated one.

delcroip commented on setup.py:
```diff
@@ -1,45 +1,45 @@
-    name='openimis-be-claim',
-    version='1.4.0',
+    name="openimis-be-claim",
+    version="1.4.0",
```
potentially problematic with the publish action TBC

delcroip committed:
Delete t.patch

delcroip commented:
> @delcroip this is required , i still cannot test in postgres , throws error for test db. I can share full error log if required but , how about testing with test json, test objects, etc. similar to simulated one.

Normally all is ready to merge all the feature/test-empty-db PRs ; I am creating them + adding a comment on the change

delcroip commented:
other similar branches https://github.com/openimis/openimis-be_py/blob/feature/test-empty-db/openimis.json

delcroip committed:
chore: avoid 2x same error

delcroip committed:
refactor(claim): optimize claim history query to use legacy_id subquery

- Removed unnecessary claim object fetch in schema.py, replacing code-based filter with subquery on legacy_id for better performance and accuracy.
- Updated gql_queries.py with minor formatting.
- Refactored tests in tests.py to simulate claim history via status changes and processing, eliminating redundant claim creation.

delcroip committed:
chore: test pl use hf iso hfid

delcroip committed:
chore: rework create test price list

delcroip committed:
fix(claim): resolve merge conflicts in validation and service tests

Resolved merge conflicts in claim/tests/test_validations.py and claim/tests/tests_services.py to integrate recent changes from the backend branch, ensuring test consistency and preventing integration issues. Changes include reconciled assertions and test setups without altering core functionality.
